### ISSUE TITLE
Add extended demo launch file with obstacles and EKF filtering

### ISSUE THREAD

**Issue by oscardegroot** [lassepe/motion-planning-hackathon#2]
Since most of us implemented something to read obstacles and apply EKF filtering to get their velocities. I think it might be nice to add an `demo_with_obstacles.launch` file where people can specify what Vicon objects should be read as obstacles and it runs an EKF for each to get their velocity.

What do you think @lassepe? I can implement it, if you agree that it is useful ;)

**Comment by lassepe** [comment]
I agree this is useful and I would appreciate a contribution. You have to be a bit careful when you do that: in our existing EKF config, `robot_localization` publishes `map->odom` and assumes existence of `map->baseline`. I think there should be options in `robot_localization` to use different frames (and not rely on the existence of `odom->obstacle`) but it involves some digging. I suspect that you would have to set the `odom` frame in the ekf.yaml to `map` and the baselink frame to `obstacle` or something

**Comment by oscardegroot** [comment]
Okay, I will look into it. 

For the obstacles, is it actually necessary to set the tf frames? I used `publish_tf: false` yesterday to avoid it from publishing any tfs (it just publishes the odometry on the associated topic).

**Comment by lassepe** [comment]
> For the obstacles, is it actually necessary to set the tf frames? I used publish_tf: false yesterday to avoid it from publishing any tfs (it just publishes the odometry on the associated topic).

Yes, that should work; so long as you make sure it doesn't try to use `odom->baselink` (since it may otherwise use the robots odom for the obstacles estimate)
